### PR TITLE
Add decomposition for aten.roll

### PR DESCRIPTION
**Pull request by fortianyou**
1. Add `aten.roll` to torch_ods_gen.py
2. Add decomposition for aten.roll
3. Add decomposition filecheck for aten.roll
4. Add a e2e unit test

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -709,6 +709,62 @@ class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
 };
 } // namespace
 
+// Decompose aten.roll into aten.expand and aten.slice and aten.cat ops.
```
Comment (pashu123) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:712]: Nit: aten.expand, aten.slice and aten.cat ops.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -709,6 +709,62 @@ class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
...
+    Value constZero = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
+    Value constOne = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    auto self = op.self();
+    Type listType = Torch::ListType::get(self.getType());
+    // roll(input, shift, dim) = cat({
+    //   slice(input, dim, -shift, none),
+    //   slice(input, dim, 0, -shift)}, dim)
+    auto ImitateRoll = [&](Value input, Value shift, Value dim) {
```
Comment (pashu123) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:743]: Nit: imitateRoll

**Comment by pashu123** [comment]
@fortianyou Please add an e2e example.

**Comment by silvasean** [comment]
Hey MHLO folks, before you add too many new ops, can you please plug in the MHLO backend into the end-to-end tests in our CI? More info here: https://github.com/llvm/torch-mlir/pull/1025#issuecomment-1178465875

I'm really worried about adding many new ops without our e2e tests.

**Comment by fortianyou** [comment]
> can you please plug in the MHLO backend into the end-to-end tests in our CI

Thanks for reminding me @silvasean. I will try to figure out how to add the e2e tests for MHLO in Torch-MLIR CI.

But the PR only added codes to decomposition, and I tried to add e2e UT in the lowering pass of linalg-on-tensors

**Comment by fortianyou** [comment]
@pashu123 Can you have a glance at my code? Why is the data type of `torch.roll` output unknown in the e2e test?
image

**Comment by pashu123** [comment]
> @pashu123 Can you have a glance at my code? Why is the data type of `torch.roll` output unknown in the e2e test? image

You will have to add the out type in refine types pass. https://github.com/llvm/torch-mlir/blob/bb47c166a047ddbc913950cb0f2571594c0fe261/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L661

**Comment by fortianyou** [comment]
@pashu123 @silvasean After rebasing the latest main branch codes, all the tests are passed. Do you have any other comments?

**Comment by silvasean** [comment]
LGTM! Please make sure to squash all your commits before merging.

**Pull request merged**